docs/Plugin Architecture.md

```diff
@@ -0,0 +1,70 @@
+## Creating
+
+Package names may only contain lowercase letters and hyphens to separate words. To scaffold out a new package:
+
+```fish
+$ omf new pkg my_package
+
+my_package/
+  README.md
+  my_package.fish
+  completions/my_package.fish
+```
+
+> Use `omf new theme my_theme` for themes.
+
+Please provide [auto completion](http://fishshell.com/docs/current/commands.html#complete) for your utilities if applicable and describe how your package works in the `README.md`.
+
+
+`my_package.fish` defines a single function:
+
+```fish
+function my_package -d "My package"
+end
+```
+
+> Bear in mind that fish lacks a private scope so consider the following options to avoid polluting the global namespace:
+
++ Prefix functions: `my_package_my_func`.
```

Comment:
`my_package_my_func` -> `my_package.my_func`. You may also want to link to the style guide here.